### PR TITLE
fix(cli): fix bug that required cdktf.json in parent directory when running convert

### LINKED ISSUE #1247: Convert function cdktf-cli always errors with Maximum call stack size exceeded
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

**cdktf-cli version:** 0.7.0
**Terraform version:** 0.15.5
**Python version:** 3.9

### Affected Resource(s)

`cdktf convert` command fails with the error `Maximum call stack size exceeded` whatever the file used maybe. Downgrading the CLI to 0.6.4 did not help. Upgrading to the latest development version also did not help.

### Debug Output

https://gist.github.com/nandac/7f7522a6e0de22ea01f6f5428943c7df

### Expected Behavior
Converted CDKTF content should either be available in the file provided or printed to stdout.

### Actual Behavior
Errors out with `Maximum call stack size exceeded` with output provided in the gist.

### Steps to Reproduce
Assuming `versions.tf` contains the following content:
```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }

    external = {
      source  = "hashicorp/external"
      version = "~> 2.0"
    }

    local = {
      source  = "hashicorp/local"
      version = "~> 2.0"
    }

    null = {
      source  = "hashicorp/null"
      version = "~> 3.0"
    }

    spotinst = {
      source  = "spotinst/spotinst"
      version = "~> 1.0"
    }

    template = {
      source  = "hashicorp/template"
      version = "~> 2.2"
    }
  }

  required_version = "0.15.5"
}
```

On the console I tried:
`cat versions.tf | cdktf convert --provider hashicorp/aws --language python`

### References

- #0000

### PR DESCRIPTION
**Pull request by ansgarm**
`path.resolve(rootPath, ..)` will return rootPath if there's no more parent directory.
This caused an infinite loop ending in `RangeError: Maximum call stack size exceeded`.

Resolves #1247

### CODE REVIEW

**Review by skorfmann — APPROVED** [review]

File: packages/cdktf-cli/bin/cmds/convert.ts [hunk 1]
```diff
@@ -37,8 +37,9 @@ function findFileAboveCwd(
     return fullPath;
   }
 
-  if (fs.existsSync(path.resolve(rootPath, ".."))) {
-    return findFileAboveCwd(file, path.resolve(rootPath, ".."));
+  const parentDir = path.resolve(rootPath, "..");
+  if (fs.existsSync(parentDir) && parentDir !== rootPath) {
```
Comment (DanielMSchmidt) [inline packages/cdktf-cli/bin/cmds/convert.ts:41]: I think a parentDir !== rootPath => we reached the top comment would be nice here

**Review by DanielMSchmidt — APPROVED** [review]

File: packages/cdktf-cli/bin/cmds/convert.ts (hunk 1 above)
Comment (ansgarm) [reply packages/cdktf-cli/bin/cmds/convert.ts:41]: Damn, too late 😅

Comment (ansgarm) [reply packages/cdktf-cli/bin/cmds/convert.ts:41]: But you're right

Comment (ansgarm) [reply packages/cdktf-cli/bin/cmds/convert.ts:41]: The commit message contains that info, however 😁

Comment (ansgarm) [reply packages/cdktf-cli/bin/cmds/convert.ts:41]: Should I still follow-up?

Comment (DanielMSchmidt) [reply packages/cdktf-cli/bin/cmds/convert.ts:41]: Nah, I think the commit message should be enough

**Pull request merged**